db/dcrpg/queries.go

```diff
@@ -356,15 +362,63 @@ func InsertVotes(db *sql.DB, dbTxns []*dbtypes.Tx, _ /*txDbIDs*/ []uint64, fTx *
 		return nil, nil, nil, nil, nil, err
 	}
 
+	// Prepare agenda votes insert statement.
+	agendaVotesInsert := internal.MakeAgendaVotesInsertStatement(checked)
+	agendaVotesStmt, err := dbtx.Prepare(agendaVotesInsert)
+	if err != nil {
+		log.Errorf("Agenda Votes INSERT prepare: %v", err)
+		_ = voteStmt.Close()
+		_ = dbtx.Rollback() // try, but we want the Prepare error back
+		return nil, nil, nil, nil, nil, err
+	}
+
 	bail := func() {
 		// Already up a creek. Just log any Rollback error.
 		_ = voteStmt.Close()
 		_ = agendaStmt.Close()
+		_ = agendaVotesStmt.Close()
 		if errRoll := dbtx.Rollback(); errRoll != nil {
 			log.Errorf("Rollback failed: %v", errRoll)
 		}
 	}
 
+	// Attempt to update the storedAgenda map is help keep track of the
+	// lockedIn, hardForked and activated heights.
+	if len(storedAgendas) == 0 {
+		var id int64
+		storedAgendas, err = retrieveAllAgendas(db)
```

Comment:
All this needs commenting to express the intend.  This line would be like

```go
// Attempt to retrieve agendas from the database.
```

Reply:
I phrased it this way because at this stage it's acceptable to have *no* agendas stored in the DB.  Part of what goes on below is to store missing agendas in the DB, and update the storedAgendas cache.  I feel strongly that the intent of the code needs to be documented more here.